scripts/linters/custom_eslint_checks/rules/protractor-practices.spec.js

```diff
@@ -134,7 +140,18 @@ ruleTester.run('protractor-practices', rule, {
       code:
       `genderList.filter(function() {
         gender = 'female';
-      })`,
+      });`,
+      errors: [{
+        message: 'Please do not use .filter(), consider using a for loop',
+      }]
+    },
+    {
+      code:
+      `
+      function gradeList(name, gradebook){
+        return gradebook[name]
+      }
+      gradeList.filter(gradeList(name, gradebook));`,
```

Comment:
I think this is a more realistic use of `filter`:

```suggestion
      gradeList.filter((name) => gradeList(name, gradebook));`,
```